src/main/java/org/broadinstitute/hellbender/engine/MultiFeatureWalker.java

```diff
@@ -0,0 +1,246 @@
+     * dictionaries available.)
+     * This method also concatenates the sample names available from each feature input.
+     */
+    private void setDictionaryAndSamples() {
+        dictionary = getMasterSequenceDictionary();
+        if ( hasReference() ) {
+            dictionary = bestDictionary(reference.getSequenceDictionary(), dictionary);
+        }
+        if ( hasReads() ) {
+            dictionary = bestDictionary(reads.getSequenceDictionary(), dictionary);
+        }
+        for ( final FeatureInput<? extends Feature> input : features.getAllInputs() ) {
+            final Object header = features.getHeader(input);
+            if ( header instanceof SVFeaturesHeader ) {
+                final SVFeaturesHeader svFeaturesHeader = (SVFeaturesHeader)header;
+                dictionary = bestDictionary(svFeaturesHeader.getDictionary(), dictionary);
+                final List<String> sampleNames = svFeaturesHeader.getSampleNames();
+                if ( sampleNames != null ) {
+                    samples.addAll(svFeaturesHeader.getSampleNames());
+                }
+            } else if (header instanceof VCFHeader ) {
+                final VCFHeader vcfHeader = (VCFHeader)header;
+                dictionary = bestDictionary(vcfHeader.getSequenceDictionary(), dictionary);
+                samples.addAll(vcfHeader.getSampleNamesInOrder());
+            }
+        }
+        if ( dictionary == null ) {
+            throw new UserException("No dictionary found.  Provide one as --" +
+                    StandardArgumentDefinitions.SEQUENCE_DICTIONARY_NAME + " or --" +
+                    StandardArgumentDefinitions.REFERENCE_LONG_NAME + ".");
+        }
+    }
+
+    private SAMSequenceDictionary bestDictionary( final SAMSequenceDictionary newDict,
+                                                    final SAMSequenceDictionary curDict ) {
+        if ( curDict == null ) return newDict;
+        if ( newDict == null ) return curDict;
+        final SAMSequenceDictionary smallDict;
+        final SAMSequenceDictionary largeDict;
+        if ( newDict.size() <= curDict.size() ) {
+            smallDict = newDict;
+            largeDict = curDict;
+        } else {
+            smallDict = curDict;
+            largeDict = newDict;
+        }
+        int lastIdx = -1;
+        for ( final SAMSequenceRecord rec : smallDict.getSequences() ) {
+            final int newIdx = largeDict.getSequenceIndex(rec.getContig());
+            if ( newIdx == -1 ) {
+                throw new UserException("Contig " + rec.getContig() +
+                                        " not found in the larger dictionary");
+            }
+            if ( newIdx <= lastIdx ) {
+                throw new UserException("Contig " + rec.getContig() +
+                                        " not in same order as in larger dictionary");
+            }
+            lastIdx = newIdx;
```

Comment:
Can't use SequenceDictionaryUtils.supersets().  That method doesn't test whether the like-named contigs have the same order.  We're using the dictionary to get the collation sequence so that we can present the features in sorted order.  It's the often revisited problem that Features, SimpleIntervals, etc. aren't comparable.